state/runtime/evm/instructions_test.go

```diff
@@ -90,3 +94,285 @@ func TestMStore(t *testing.T) {
 
 	assert.Len(t, s.memory, 1024+32)
 }
+
+type mockHostForCreate struct {
+	mockHost
+	nonce       uint64
+	callxResult *runtime.ExecutionResult
+}
+
+var _ runtime.Host = (*mockHostForCreate)(nil)
+
+func (m *mockHostForCreate) GetNonce(types.Address) uint64 {
+	return m.nonce
+}
+
+func (m *mockHostForCreate) Callx(*runtime.Contract, runtime.Host) *runtime.ExecutionResult {
+	return m.callxResult
+}
+
+var (
+	addr1 = types.StringToAddress("1")
+)
+
+func TestCreate(t *testing.T) {
+	type state struct {
+		gas    uint64
+		sp     int
+		stack  []*big.Int
+		memory []byte
+		stop   bool
+		err    error
+	}
+
+	addressToBigInt := func(addr types.Address) *big.Int {
+		return new(big.Int).SetBytes(addr[:])
+	}
+
+	tests := []struct {
+		name        string
+		op          OpCode
+		contract    *runtime.Contract
+		config      *chain.ForksInTime
+		initState   *state
+		resultState *state
+		mockHost    *mockHostForCreate
+	}{
+		{
+			name: "should succeed in case of CREATE",
+			op:   CREATE,
+			contract: &runtime.Contract{
+				Static:  false,
+				Address: addr1,
+			},
+			config: &chain.ForksInTime{},
+			initState: &state{
+				gas: 1000,
+				sp:  3,
+				stack: []*big.Int{
+					big.NewInt(0x01), // length
+					big.NewInt(0x00), // offset
+					big.NewInt(0x00), // value
+				},
+				memory: []byte{
+					byte(REVERT),
+				},
+			},
+			resultState: &state{
+				gas: 500,
+				sp:  1,
+				stack: []*big.Int{
+					addressToBigInt(crypto.CreateAddress(addr1, 0)), // contract address
+					big.NewInt(0x00),
+					big.NewInt(0x00),
+				},
+				memory: []byte{
+					byte(REVERT),
+				},
+			},
+			mockHost: &mockHostForCreate{
+				nonce: 0,
+				callxResult: &runtime.ExecutionResult{
+					GasLeft: 500,
+					GasUsed: 500,
+				},
+			},
+		},
+		{
+			name: "should throw errWriteProtection in case of static call",
+			op:   CREATE,
+			contract: &runtime.Contract{
+				Static: true,
+			},
+			config: &chain.ForksInTime{},
+			initState: &state{
+				gas: 1000,
+				sp:  3,
+				stack: []*big.Int{
+					big.NewInt(0x01), // length
+					big.NewInt(0x00), // offset
+					big.NewInt(0x00), // value
+				},
+				memory: []byte{
+					byte(REVERT),
+				},
+				stop: false,
+				err:  nil,
+			},
+			// shouldn't change any states except for stop and err
+			resultState: &state{
+				gas: 1000,
+				sp:  3,
+				stack: []*big.Int{
+					big.NewInt(0x01), // length
+					big.NewInt(0x00), // offset
+					big.NewInt(0x00), // value
+				},
+				memory: []byte{
+					byte(REVERT),
+				},
+				stop: true,
+				err:  errWriteProtection,
+			},
+			mockHost: &mockHostForCreate{},
+		},
+		{
+			name:     "should throw errOpCodeNotFound when op is CREATE2 and config.Constantinople is not disabled",
```

Comment:
I think you have a typo here:
```suggestion
			name:     "should throw errOpCodeNotFound when op is CREATE2 and config.Constantinople is disabled",
```

Reply:
Fixed